cmd/build/v2/build.go

```diff
@@ -167,56 +180,92 @@ func (bc *OktetoBuilder) Build(ctx context.Context, options *types.BuildOptions)
 
 			buildSvcInfo := buildManifest[svcToBuild]
 
+			// create the meta pointer and append it to the analytics slice
+			meta := analytics.NewImageBuildMetadata()
+			buildsAnalytics = append(buildsAnalytics, meta)
+
+			meta.Name = svcToBuild
+			meta.RepoURL = bc.Config.GetAnonymizedRepo()
+
+			repoHashDurationStart := time.Now()
+			repoCommit := bc.Config.GetGitCommit()
+			buildHash := getBuildHashFromCommit(buildSvcInfo, repoCommit)
+
+			meta.RepoHash = buildHash
+			meta.RepoHashDuration = time.Since(repoHashDurationStart)
+
 			// We only check that the image is built in the global registry if the noCache option is not set
 			if !options.NoCache && bc.Config.IsCleanProject() {
+
 				imageChecker := getImageChecker(buildSvcInfo, bc.Config, bc.Registry)
-				if imageWithDigest, isBuilt := imageChecker.checkIfCommitHashIsBuilt(options.Manifest.Name, svcToBuild, buildSvcInfo); isBuilt {
-					oktetoLog.Information("Skipping build of '%s' image because it's already built for commit %s", svcToBuild, bc.Config.GetGitCommit())
+				cacheHitDurationStart := time.Now()
+				imageWithDigest, isBuilt := imageChecker.checkIfBuildHashIsBuilt(options.Manifest.Name, svcToBuild, buildHash)
+
+				meta.CacheHit = isBuilt
+				meta.CacheHitDuration = time.Since(cacheHitDurationStart)
+
+				if isBuilt {
+					oktetoLog.Information("Skipping build of '%s' image because it's already built for commit %s", svcToBuild, repoCommit)
 					// if the built image belongs to global registry we clone it to the dev registry
 					// so that in can be used in dev containers (i.e. okteto up)
 					if bc.Registry.IsGlobalRegistry(imageWithDigest) {
 						oktetoLog.Debugf("Copying image '%s' from global to personal registry", svcToBuild)
-						tag := bc.Config.GetBuildHash(buildSvcInfo)
+						tag := buildHash
 						devImage, err := bc.Registry.CloneGlobalImageToDev(imageWithDigest, tag)
 						if err != nil {
+							bc.analyticsTracker.TrackImageBuild(buildsAnalytics...)
 							return err
 						}
 						imageWithDigest = devImage
 					}
 
 					bc.SetServiceEnvVars(svcToBuild, imageWithDigest)
-					bc.builtImages[svcToBuild] = true
+					builtImagesControl[svcToBuild] = true
+					meta.Success = true
 					continue
 				}
 			}
 
 			if !okteto.Context().IsOkteto && buildSvcInfo.Image == "" {
+				bc.analyticsTracker.TrackImageBuild(buildsAnalytics...)
 				return fmt.Errorf("'build.%s.image' is required if your cluster doesn't have Okteto installed", svcToBuild)
 			}
-
+			buildDurationStart := time.Now()
```

Comment:
same as before, here we don't know what is inside `buildService` func. If we need to check build duration I would track the time we used until external lib call finish the build

Reply:
I've added a comment to the function. This is where the built is done, either a single one, or 2 due to the volume mounts. 

If an error occurs the analytics does not include buld time (no build)

@pchico83 what should we track when building with volume mounts? should perhaps a boolean that indicates that volumes have been added? so the time would be of 2 builds?

Reply:
@teresaromero ideally, it should be a single event. but if it simplifies the logic, we can track two independent build events. Or ignore the builds due to volume mounts

Reply:
I've been going though this and right now it will trigger a single event.

There are cases where there is just 1 build that is with the volume mounts, or just 1 build with a dockerfile, or both one after the other...
I think its valueable to have the event to track the build time, and a property that also inform if is with volumes or not to be able to distinguish between building times....